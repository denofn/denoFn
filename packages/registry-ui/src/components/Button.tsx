import React from "react";
import { css } from "otion";
import { Spinner } from "./Spinner";

export type ButtonProps = {
  className?: string;
  loading?: boolean;
  title: string;
  disabled?: boolean;
  onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
};

export const Button = React.memo(function Button({ className, loading, disabled, title, onClick }: ButtonProps) {
  return (
    <button
      className={`${css({
        backgroundColor: "transparent",
        fontWeight: "bold",
        padding: ".5rem 1rem",
        borderWidth: "1px",
        borderStyle: "solid",
        borderColor: "rgb(26, 32, 44)",
        borderRadius: ".25rem",
        lineHeight: "1rem",
        fontSize: "1rem",
        cursor: "pointer",
        ":hover": {
          backgroundColor: "rgb(26, 32, 44) !important",
          color: "white",
          borderColor: "transparent",
        },
        ":disabled": {
          color: loading ? "rgb(26, 32, 44)" : "rgba(26,32,44,.5)",
          borderColor: loading ? "rgb(26, 32, 44)" : "rgba(26,32,44,.5)",
          cursor: "not-allowed",
        },
        selectors: {
          "&:disabled:hover": {
            backgroundColor: "transparent !important",
            borderColor: loading ? "rgb(26, 32, 44)" : "rgba(26,32,44,.5)",
          },
        },
      })}${className ? ` ${className}` : ""}`}
      disabled={disabled ?? false}
      onClick={onClick}
    >
      <div
        className={css({
          display: "flex",
          alignItems: "center",
        })}
      >
        {loading && (
          <span className={css({ marginRight: ".33rem" })}>
            <Spinner dim=".66rem" thickness=".125rem" />
          </span>
        )}
        {title}
      </div>
    </button>
  );
});
