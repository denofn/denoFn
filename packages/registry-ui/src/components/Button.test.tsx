import { render, cleanup } from "@testing-library/react";
import React from "react";

import { Button } from "./Button";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<Button title="title" onClick={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
