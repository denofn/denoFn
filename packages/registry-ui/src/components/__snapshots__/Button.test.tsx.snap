// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <button
    class="_1peqg37 _1efi8gv _1k9emg3 _my9kfw _fxc8cu _g3a0px _sy4y6p _r24itg _rkiwex _e0dnmk _1ijvbgw _gxlvbl _ryz482 _yoqe5q _141w5m3 _1y0fyy9 _1g8t4a7 _1n02zsb"
  >
    <div
      class="_zjik7 _1h3rtzg"
    >
      title
    </div>
  </button>
</DocumentFragment>
`;
